rust/catalyst-types/src/id_uri/role_index.rs

```diff
@@ -12,54 +12,82 @@ use thiserror::Error;
 /// Role Index parsing error
 #[derive(Display, Error, Debug)]
 #[allow(clippy::module_name_repetitions)]
-pub enum RoleIndexError {
+pub enum RoleIdError {
+    /// Invalid parse
+    Parse(#[from] ParseIntError),
+
     /// Invalid Role Index
-    InvalidRole(#[from] ParseIntError),
+    InvalidRole(u8),
 }
 
 /// Project Catalyst User Role Index.
 ///
 /// <https://github.com/input-output-hk/catalyst-CIPs/blob/x509-catalyst-role-definitions/CIP-XXXX/README.md>
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct RoleIndex(u16);
-
-impl RoleIndex {
-    /// Delegated Representative
-    pub const DREP: RoleIndex = RoleIndex(1);
-    /// Proposer
-    pub const PROPOSER: RoleIndex = RoleIndex(3);
-    /// Default Role 0
-    pub const ROLE_0: RoleIndex = RoleIndex(0);
+#[repr(u8)]
+#[non_exhaustive]
+pub enum RoleId {
+    /// Can Vote and Comment on Proposals.
+    Role0 = 0,
+    /// Allows for voters to give their voting power to the dRep who then votes on their
+    /// behalf.
```

Comment:
I know this is from the doc, but this would be clearer
```suggestion
    /// Delegated representative that vote on behalf of delegators.
```